### ISSUE TITLE
Plugin not grouping users when those changes

### ISSUE THREAD

**Issue by achybajo** [emmarichardson/local_autogroup#26]
Hi! i'm new to Moodle and found your plugin, which is very accurate for a common problem.
Anyway, i'm having the following issue:

installed the plugin succesfully, activated and configured it easyly. It also created the groups for a custom field in the users profile, but when a new user signed in and completed that custom field, the group is not refreshing the data, i mean the new users whom should be in a previously created group (by autogroup) are not getting assigned to the group they should be.

Mi custom field is a combo box for "work city" of the user.

My Moodle version is: Moodle 3.9.2+ (Build: 20201002) | Versión 2020061502.04 | stable |
My autogroup plugin version is:  2.5 | local_autogroup | 2020072700

is there any check or fix to get this propperly working?? 

Thanks in advance, and sorry about my english (not native speaker here).
Regards!

**Comment by emmarichardson** [comment]
I have not tested on 3.9.2 but it is working as expected on 3.9.1…can you tell me exactly what you did?  You might try checking the membership/group membership in settings if you have made any manual changes...

Emma

**Comment by achybajo** [comment]
Hi Emma, thanks for your quick answer!.

i have installed the plugin downloading the .zip file from https://moodle.org/plugins/pluginversions.php?plugin=local_autogroup.

Once installed, selected the custom field to use (i have only this one custom field in user profile). 
Then i tried it with a student user i have to test and worked fine, when we open the registration form to the people, first time worked with new users (around 80 users). Then other 140 users registered but wasn't grouped propperly. Also the plugin created the groups without problem; nevertheless when users modify their profile (changing the value of custom field) are not  getting re-grouped propperly.

I have the following configuration for the plugin:
 
![autogroup 1](https://user-images.githubusercontent.com/30998196/95256578-618fd780-07f9-11eb-87a0-aade734dae0a.png)
![autogroup 2](https://user-images.githubusercontent.com/30998196/95256583-62c10480-07f9-11eb-95e6-3ddd06054761.png)
![autogroup 3](https://user-images.githubusercontent.com/30998196/95256584-63599b00-07f9-11eb-9c19-d54bb8764a13.png)

Hope you can give some feedback.
Regards!

**Comment by emmarichardson** [comment]
So it put the 140 in the wrong groups?  Did you make any changes between the 80 and the 140?  Added extra fields in the profile field or something?  That could mess things up.

Emma

Emma Richardson
Distance Learning Coordinator
Direct: 719-650-3037

**Comment by achybajo** [comment]
Nope, the new 140 users did not group in any pre-created group. Simply they were left unnasigned. 

Not extra fields added in the profile.. it looks like a grouping trigger to me, the data is there, profiles are right and working fine, but groups created by autogroup (there is no manually created groups) are not reflecting the exact info. some people are left without group despite of custom field being propperly complete.

It's my first time using the plugin and administrating moodle, hope not doing it too bad! 
Thank you Emma!!

**Comment by emmarichardson** [comment]
Do they have one of the correct roles in the course?

Emma

Emma Richardson
Distance Learning Coordinator
Direct: 719-650-3037

**Comment by achybajo** [comment]
Everyone is student in the course, because are using auto-registration to the course.

we did not filter or aprove the students manually, every person registers to their propper course and automatically gets the role student.

Wish to had understood your question!
Thanks again!

**Comment by emmarichardson** [comment]
Could you try turning full debugging on and then enrolling as a student and see if there are any error messages that come up...

Emma

Emma Richardson
Distance Learning Coordinator
Direct: 719-650-3037

**Comment by achybajo** [comment]
of course! will do tomorrow and get back to you with the news..

Thanks for your time!
